### ISSUE TITLE
BasePacketReader is badly-behaved if read_packet is not continually drained

### ISSUE THREAD

**Issue by Ralith** [est31/ogg#28]
Pushing every page of a complete ogg file before attempting to read any packets will result in lost packets.

**Comment by est31** [comment]
Yeah there is only limited buffering. Maybe this should be pointed out.

**Comment by Ralith** [comment]
It is very convenient when coupling push-based and pull-based interfaces to push all your data before trying to drain packets, so having undocumented breakage there is definitely an unpleasant surprise.
